### PR TITLE
Nadro/adhoc/revolve api improvement

### PR DESCRIPTION
**Pull request by nadr0**
relates https://github.com/KittyCAD/modeling-app/issues/2540

- `-Z` and `Z` are not allowed as revolve axis arguments. The revolve only happens on the local sketch plane of `Y/-Y` and `X/-X`. 
- Angle is defined by `[0.0,360.0]` (inclusive)
- Added another sentence to clarify in the documentation that revolve happens around a local sketch axis and not a global axis

### CODE REVIEW

File: src/wasm-lib/kcl/src/std/revolve.rs [hunk 1]
```diff
@@ -255,10 +247,10 @@ async fn inner_revolve(
     args: Args,
 ) -> Result<Box<ExtrudeGroup>, KclError> {
     if let Some(angle) = data.angle {
-        // Return an error if the angle is less than -360 or greater than 360.
-        if !(-360.0..=360.0).contains(&angle) {
+        // Return an error if the angle is less than -360 or greater than 360 and cannot be 0.0
+        if !(-360.0..=360.0).contains(&angle) || angle == 0.0 {
```
Comment (jtran) [inline src/wasm-lib/kcl/src/std/revolve.rs:251]: Should this do an epsilon comparison? I.e. also error if it's within epsilon of zero? As for what the value of epsilon should be... I did a quick search of modeling-app, and I didn't find anything that looks like what we want. Might need to see what the engine needs.

Comment (jessfraz) [reply src/wasm-lib/kcl/src/std/revolve.rs:251]: we can validate this with the same library we use in settings https://docs.rs/validator/latest/validator/ with a derive, makews it easier and it integrates into json scehma so the docs will auto get it, its win win

Comment (jessfraz) [reply src/wasm-lib/kcl/src/std/revolve.rs:251]: happy to pair if you need help

File: docs/kcl/revolve.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ layout: manual
 
 Rotate a sketch around some provided axis, creating a solid from its extent.
 
-This, like extrude, is able to create a 3-dimensional solid from a 2-dimensional sketch. However, unlike extrude, this creates a solid by using the extent of the sketch as its revolved around an axis rather than using the extent of the sketch linearly translated through a third dimension.
+This, like extrude, is able to create a 3-dimensional solid from a 2-dimensional sketch. However, unlike extrude, this creates a solid by using the extent of the sketch as its revolved around an axis rather than using the extent of the sketch linearly translated through a third dimension. Revolve occurs around a local sketch axis rather than a global axis.
```
Comment (jessfraz) [inline docs/kcl/revolve.md:9]: we should expose is_axis_2d to the user, in like a optional param "global" and then they can override this local only thing see discussion here: https://github.com/KittyCAD/modeling-app/issues/3959

Comment (jessfraz) [reply docs/kcl/revolve.md:9]: okay nevermind kurt says dont pull that thru and has valid reasons in that issue, you could fix the plane stuff he mentioned if you want but i also dont mind doing a pr for that

Comment (jessfraz) [reply docs/kcl/revolve.md:9]: didn't know how dirty you wanted to get into the rust side

File: src/wasm-lib/kcl/src/std/revolve.rs (hunk 1 above)
Comment (nadr0) [reply src/wasm-lib/kcl/src/std/revolve.rs:251]: I talked to Adam about this in Rust Club, we ended up opting for using the same pattern in `inner_revolve`. My first implementation of the `validator` did not work as expected. Also I think when we tried doing the `validator` method in rust club it had an issue? Then we thought about making the `Option<f64>` into an `Option<RevolveAngle>` struct that would do the verification?

Yeah, I'll pair if you want!

Comment (jessfraz) [reply src/wasm-lib/kcl/src/std/revolve.rs:251]: okay yeah if we use validator here we get weirder errors but we can at least add the range to the schemars like so: https://github.com/KittyCAD/modeling-app/commit/23cd16a2329199db43120c9d17f82c1154ebe418#diff-b4e1769ab46a04e0d1bde955e84efb6360ca5d8165dcbe3f9261f572021fd0e3

Comment (jessfraz) [reply src/wasm-lib/kcl/src/std/revolve.rs:251]: I added it to arc too, we dont currently use it for the markdown docs but at least its in the generated schema

**Comment by nadr0** [comment]
Closing this since this was resolved from two separate other PRs. Revolve axis was updated as well as the min max range.

**Pull request closed without merging**